commit b7d8cce5b558e0c0aa6898c9865356481598b46d
Author: Alex Deucher <alexdeucher@gmail.com>
Date:   Mon Oct 25 19:44:00 2010 -0400

    drm/radeon/kms: MC vram map needs to be >= pci aperture size
    
    The vram map in the radeon memory controller needs to be
    >= the pci aperture size.  Fixes:
    https://bugs.freedesktop.org/show_bug.cgi?id=28402
    
    The problematic cards in the above bug have 64 MB of vram,
    but the pci aperture is 128 MB and the MC vram map was only
    64 MB.  This can lead to hangs.
    
    Signed-off-by: Alex Deucher <alexdeucher@gmail.com>
    Cc: stable@kernel.org
    Signed-off-by: Dave Airlie <airlied@redhat.com>

diff --git a/drivers/gpu/drm/radeon/r100.c b/drivers/gpu/drm/radeon/r100.c
index 6112ac9..6d1540c 100644
--- a/drivers/gpu/drm/radeon/r100.c
+++ b/drivers/gpu/drm/radeon/r100.c
@@ -2270,6 +2270,9 @@ void r100_vram_init_sizes(struct radeon_device *rdev)
 		/* Fix for RN50, M6, M7 with 8/16/32(??) MBs of VRAM - 
 		 * Novell bug 204882 + along with lots of ubuntu ones
 		 */
+		if (rdev->mc.aper_size > config_aper_size)
+			config_aper_size = rdev->mc.aper_size;
+
 		if (config_aper_size > rdev->mc.real_vram_size)
 			rdev->mc.mc_vram_size = config_aper_size;
 		else
